pkg/vault/envconsul/webhook.go

```diff
@@ -0,0 +1,177 @@
+
+	webhookConfig := &corev1.ConfigMap{}
+	i.client.Get(ctx, i.webhookConfigKey, webhookConfig)
+	image, _ := webhookConfig.Data["image"]
+	installPath, _ := webhookConfig.Data["install_path"]
+
+	containerConfigPaths := getContainerConfigPaths(config)
+	copy := pod.DeepCopy()
+
+	for k, container := range copy.Spec.Containers {
+		if configFile, ok := containerConfigPaths[container.Name]; ok {
+			c := &copy.Spec.Containers[k]
+			originalCommand := strings.Join(append(c.Command, c.Args...), " ")
+			c.Command = []string{
+				path.Join(installPath, "theatre-envconsul"),
+				"exec",
+				fmt.Sprintf("--vault-address=%s", vaultAddress),
+			}
+			if installPath != "" {
+				c.Command = append(c.Command, fmt.Sprintf("--install-path=%s", installPath))
+			}
+			if vaultAuthMountPath != "" {
+				c.Command = append(c.Command, fmt.Sprintf("--auth-backend-mount-path=%s", vaultAuthMountPath))
+			}
+			if vaultAuthRole != "" {
+				c.Command = append(c.Command, fmt.Sprintf("--auth-backend-role=%s", vaultAuthRole))
+			}
+			if configFile != "" {
+				podLogger.Log("msg", fmt.Sprintf("mutating %s with config file: %s", container.Name, configFile))
+				c.Command = append(c.Command, fmt.Sprintf("--config-file=%s", configFile))
+			} else {
+				podLogger.Log("msg", fmt.Sprintf("mutating %s without config file", container.Name))
+			}
+			c.Command = append(c.Command, fmt.Sprintf("--command=%s", originalCommand))
+
+			c.VolumeMounts = append(c.VolumeMounts, getVolumeMounts(installPath)...)
+
+			copy.Spec.InitContainers = append(copy.Spec.InitContainers, getInitContainers(image, installPath)...)
+			copy.Spec.Volumes = append(copy.Spec.Volumes, getVolumes()...)
+		} else {
+			podLogger.Log("msg", fmt.Sprintf("skipping %s: container not configured", container.Name))
+		}
+	}
+
+	return admission.PatchResponse(pod, copy)
+}
+
+func getContainerConfigPaths(config string) map[string]string {
+	containersConfigPaths := map[string]string{}
+
+	containers := strings.Split(config, ",")
+	for _, container := range containers {
+		nameAndPath := strings.Split(container, ":")
+		name := strings.TrimSpace(nameAndPath[0])
+		path := ""
+		if len(nameAndPath) == 2 {
+			path = strings.TrimSpace(nameAndPath[1])
+		}
+		containersConfigPaths[name] = path
+	}
+
+	return containersConfigPaths
+}
+
+func getInitContainers(image, path string) []corev1.Container {
+	return []corev1.Container{
+		{
+			Name:         "init-vault",
+			Image:        image,
+			Command:      []string{"/theatre-envconsul", "install", "--path", path},
+			VolumeMounts: getVolumeMounts(path),
+		},
+	}
+}
+
+func getVolumeMounts(path string) []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		{
+			Name:      "vault",
+			MountPath: path,
+		},
+	}
+}
+
+func getVolumes() []corev1.Volume {
+	return []corev1.Volume{
+		{
+			Name: "vault",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{
+					Medium: corev1.StorageMediumMemory,
```

Comment:
Do we actually want these stored in memory? I would have thought it'd be better to default to the node's default, which will be SSD in our case.
No use in wasting RAM to store binaries that'll only be called once?

Reply:
Good catch, this could be wasteful given enough pods on a node!